miden-tx/src/lib.rs

```diff
@@ -15,12 +15,12 @@ use processor::{ExecutionError, RecAdviceProvider};
 
 mod compiler;
 pub use compiler::{NoteTarget, TransactionComplier};
-mod data;
-use data::DataStore;
+pub mod data;
+pub use data::DataStore;
 mod error;
 mod executor;
 pub use error::TransactionError;
-use error::{
+pub use error::{
     DataStoreError, TransactionCompilerError, TransactionExecutorError, TransactionProverError,
     TransactionVerifierError,
 };
```

Comment:
Do we need to export the whole `data` module? In general, I'd prefer as narrow exports as possible, and I think in this case, we can probably export just `data::DataStore`.

Overall, I'd clean up a bit these exports as follows:
```Rust
mod compiler;
pub use compiler::{NoteTarget, TransactionComplier};

mod executor;
pub use executor::TransactionExecutor;

mod prover;
pub use prover::TransactionProver;

mod verifier;
pub use verifier::TransactionVerifier;

mod data;
pub use data::DataStore;

pub mod error;
```

Reply:
Perfect. That would be as narrow as possible, right? Not to import the whole `error` module.

```
mod compiler;
pub use compiler::{NoteTarget, TransactionComplier};

mod executor;
pub use executor::TransactionExecutor;

mod prover;
pub use prover::TransactionProver;

mod verifier;
pub use verifier::TransactionVerifier;

mod data;
pub use data::DataStore;

mod error;
pub use error::{
    DataStoreError, TransactionCompilerError, TransactionExecutorError, TransactionError, TransactionProverError,
    TransactionVerifierError,
};
```

Reply:
Yep - this works. Though, in this case, exposing the `error` module would have been fine too as we now re-export a lot of error types from the root namespace.